### PR TITLE
Container release policy

### PR DESCRIPTION
**Pull request by brianhlin**
First pass at a Container Release Policy doc but I think there's a few things I want to discuss/I think are missing. Also renamed and moved the Software Release Policy doc.

1. Do we really need to timestamp tag our images? CentOS just uses labels to indicate a build timestamp. I think the only difference is that we would lose some reproducibility (i.e. can't run historical images) 
1. If we do keep the timestamp tags, what's an appropriate cleanup cycle? As it's written, we'd cleanup timestamp tags based on release series but that seems a little long. It'd make cleanup easier, though.
1. Notifying users upon each build tagged as `development` lines up better with our community testing policy but seems like it'd be overly noisy.
1. How can we automatically sign our builds?

### CODE REVIEW

File: docs/policy/container-release-policy.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+This document contains policy information for container images distributed by the OSG Software Team.
+
+Contents and Sources
+--------------------
+
+Similar to our existing RPM infrastructure, container image sources, build logs, and artifacts will be stored in
+publicly available repositories (e.g. GitHub, DockerHub) for collaboration and traceability.
+Additionally, container images distributed by the OSG Software team will be based off of the latest version of a 
+[supported platform](https://opensciencegrid.org/docs/release/supported_platforms/) with software installed from OS,
+EPEL, and [osg-development](/policy/software-releases#yum-repositories) Yum repositories.
```
Comment (matyasselmeci) [inline docs/policy/container-release-policy.md]: Why development?

**Comment by matyasselmeci** [comment]
I don't think the timestamp tags are necessary. We added those for the wn-client so we could force rebuilds without making any changes; the timestamps themselves IMO are not informative because it's next to impossible to get the state of repos at a given time in the past. Dumping the `rpm -qa` output somewhere might be useful though...

File: docs/policy/container-release-policy.md (hunk 1 above)
Comment (brianhlin) [reply docs/policy/container-release-policy.md]: One of the drivers for releasing container images in the first place is to shorten the software development lifecycle so osg-development (osg-minefield in actuality but that's an internal detail) is the logical place to pull software from.

Comment (matyasselmeci) [reply docs/policy/container-release-policy.md]: (We discussed this in person and agreed that we should only be pulling selected packages out of minefield, not the whole software stack)

**Comment by brianhlin** [comment]
@matyasselmeci and I had a hallway conversation and we came to the following conclusions:

- We think we can get away with two tags, `development` and `stable`. Builds and passes automated tests -> dev; passes acceptance testing -> stable
- We shouldn't overload `development`, maybe `bleeding`?
- We can drop timestamp tags for now but store build time in a label. We're not sure if keeping old tags around is useful in a docker paradigm: you're container isn't working? Tear it down and try the latest tag!
- I'll clarify that only some packages will be installed out of OSG development

**Comment by bbockelm** [comment]
A few questions:
- how does someone roll back to a previously working container?  That was the thought behind the time stamp tag suggestion.  The bleeding edge doesn’t work now?  Then go back to the one that worked before.
- this proposal doesn’t appear to align container releases with OSG software releases.  Is that on purpose?

Would like this to be presented to the community at the AHM.

**Comment by matyasselmeci** [comment]
If we're going to have timestamps, I'd like to have a rather aggressive cleanup policy: how about once a new OSG release is made, all timestamps before the previous release are deleted? E.g. we release 3.4.30, and delete everything before 3.4.29.

**Comment by brianhlin** [comment]
> how does someone roll back to a previously working container? That was the thought behind the time stamp tag suggestion. The bleeding edge doesn’t work now? Then go back to the one that worked before.

I would think that in this case we'd suggest they use the "testing" or "stable" release, instead. Admittedly not a great solution so I can be swayed into keeping timestamp tags around.

@matyasselmeci an aggressive timestamp removal process sounds good but potentially < 1 month sounds like it may be too much. Using XCache as an example, what if there aren't any XCache/XRootD/image updates for months? Then we'd be getting rid of those historical images.

> this proposal doesn’t appear to align container releases with OSG software releases. Is that on purpose?

I was thinking of container releases as a relatively independent process from software releases. We could have the "stable" tagging line up with releases if we retain the three tag setup but it's not clear to me that it's useful.

**Comment by matyasselmeci** [comment]
How about weekly timestamps for 3 months, then monthly timestamps for 1 year, then delete after that?

**Comment by timtheisen** [comment]
I'm not sure about the 'fast' and 'slow' tags. I think that 'tested' might be better than 'slow'. Not sure what to call 'fast', perhaps 'new'.

**Comment by edquist** [comment]
I think it's unfortunate that the term `cutting` fell into disuse after the 90s

**Comment by brianhlin** [comment]
Yeah, the tags are up for debate. 'Tested', though, gives the impression that the other tag is untested, which wouldn't be strictly true. Also, it's probably best to avoid anything we use for the Yum repositories since we don't want to conflate them. Spitballing some names:

- Fast/Nightly (...if we did nightly builds)/Bleeding
- Stable/Production

**Comment by edquist** [comment]
fwiw, `nightly` is has a pretty straightforward, honest, and commonly understood meaning

so it gets my :+1:

**Comment by timtheisen** [comment]
LibreOffice uses 'fresh' and 'still'.

Maybe 'fresh' and 'stable'

**Comment by matyasselmeci** [comment]
Perhaps this discussion would be more productive in real-time, via Slack or in person?

**Comment by brianhlin** [comment]
I'm going to merge this so we can get feedback from stakeholders. We can discuss the tag names at HOW or at a technology area meeting.

**Comment by sfiligoi** [comment]
A little late, but here is my 2c:
Picking "the latest supported OS version" has significant drawbacks.

Remember that container runtimes are backward but not forward compatible.
I.e. you can run a CentOS 6 container on a CentOS 7 host, but you should not run a CentOS 7 container on a CentOS 6 host.

**Comment by brianhlin** [comment]
That's an interesting point but I don't think it's that strict since the kernel is fairly stable. Though I can imagine cases where software depends on new kernel features that could result in tricky-to-troubleshoot issues.

When there are new OS versions, we can tag the last stable versions corresponding to the previous OS: when CentOS 8 is released, we'd tag the latest CentOS 7 image with `el7`.

**Pull request merged**